dockerfiles/cli/tests/cmd_start_stop_restart_tests.bats

```diff
@@ -0,0 +1,102 @@
+    [ "$status" -eq 1 ]
+    [ "$output" = "Default port 8080 for che server is used. Cannot run this test on default che server port" ]
+  fi
+  tmp_path="${TESTRUN_DIR}"/cli_cmd_start_with_default_params
+  echo $tmp_path
+  mkdir -p "${tmp_path}"
+
+  #WHEN
+  docker run --rm -v "${SCRIPTS_DIR}":/scripts/base -v /var/run/docker.sock:/var/run/docker.sock -v "${tmp_path}":/data -e CHE_CONTAINER=chetest $CLI_IMAGE start --skip:nightly --skip:pull
+
+  #THEN
+  [[ "$(docker inspect --format='{{.State.Running}}' chetest)" == "true" ]]
+  ip_address=$(docker inspect -f {{.NetworkSettings.Networks.bridge.IPAddress}} chetest)
+  curl -fsS http://${ip_address}:8080  > /dev/null
+}
+
+@test "test cli 'stop' command with default settings" {
+  #GIVEN
+  if [ ! port_is_free 8080 ]; then
+    [ "$status" -eq 1 ]
+    [ "$output" = "Default port 8080 for che server is used. Cannot run this test on default che server port" ]
+  fi
+  tmp_path="${TESTRUN_DIR}"/cli_cmd_stop_with_default_settings
+  echo $tmp_path
+  mkdir -p "${tmp_path}"
+  docker run --rm -v "${SCRIPTS_DIR}":/scripts/base -v /var/run/docker.sock:/var/run/docker.sock -v "${tmp_path}":/data -e CHE_CONTAINER=chetest $CLI_IMAGE start --skip:nightly --skip:pull
+  [[ "$(docker inspect --format='{{.State.Running}}' chetest)" == "true" ]]
+  ip_address=$(docker inspect -f {{.NetworkSettings.Networks.bridge.IPAddress}} chetest)
+  curl -fsS http://${ip_address}:8080  > /dev/null
+
+  #WHEN
+  docker run --rm -v "${SCRIPTS_DIR}":/scripts/base -v /var/run/docker.sock:/var/run/docker.sock -v "${tmp_path}":/data -e CHE_CONTAINER=chetest $CLI_IMAGE stop --skip:nightly --skip:pull
+
+  #THEN
+  #check that state is not set because CLI stop command also removes container.
+  [[ "$(docker inspect --format='{{.State.Running}}' chetest)" == "" ]]
+  #check that stopped container actually removed
+  [[ "$(docker ps -a)" != *"chetest"* ]]
+}
+
+@test "test cli 'restart' command with default settings" {
+  #GIVEN
+  if [ ! port_is_free 8080 ]; then
+    [ "$status" -eq 1 ]
+    [ "$output" = "Default port 8080 for che server is used. Cannot run this test on default che server port" ]
+  fi
+  tmp_path="${TESTRUN_DIR}"/cli_cmd_restart_with_default_settings
+  echo $tmp_path
+  mkdir -p "${tmp_path}"
+  docker run --rm -v "${SCRIPTS_DIR}":/scripts/base -v /var/run/docker.sock:/var/run/docker.sock -v "${tmp_path}":/data -e CHE_CONTAINER=chetest $CLI_IMAGE start --skip:nightly --skip:pull
+  [[ "$(docker inspect --format='{{.State.Running}}' chetest)" == "true" ]]
+  ip_address=$(docker inspect -f {{.NetworkSettings.Networks.bridge.IPAddress}} chetest)
+  curl -fsS http://${ip_address}:8080  > /dev/null
+
+  #WHEN
+  docker run --rm -v "${SCRIPTS_DIR}":/scripts/base -v /var/run/docker.sock:/var/run/docker.sock -v "${tmp_path}":/data -e CHE_CONTAINER=chetest $CLI_IMAGE restart --skip:nightly --skip:pull
+
+  #THEN
+  [[ "$(docker inspect --format='{{.State.Running}}' chetest)" == "true" ]]
```

Comment:
To verify that a container is running after performing restart command, I would rather choose more bulletproof check. This way, if restart would do nothing, container would still be in running state and those checks would pass.

I suggest to use e.g. .State.Pid value for check actual value of process id before and after restart. But bcs. pid is not unique, maybe better would be to use .State.StartedAt.